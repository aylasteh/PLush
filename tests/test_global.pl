
# Global variable 
var global_i : int := 5;

# Function prototpe
function putint(var in : int) : int;

# Another global variable
var global_3 : int := 3;

function main (val argc: int, val argv : [string]):int {
    var str : string;

    putchar(48 + global_i); putchar(10);

    putchar(48 + argc); putchar(10);

    var i : int := 0;

    puts(argv[0]);

    while i < argc {
        puts(argv[i]);
        i := i + 1;
    }

    putint(global_3); 
    putchar(10);

    main := 0;
}

# function definition (from prototype)
function putint(var in : int) : int {
    var out : int := in;

    if out < 10 {
        putchar(48 + in);
    }
    putint := in;
}