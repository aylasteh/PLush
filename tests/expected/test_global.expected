5
1
./test_global
./test_global
3